internal/drivers/nomad/driver.go

```diff
@@ -335,82 +351,89 @@ func (p *config) resourceJob(cpus, memGB int, vm, accountID string) (job *api.Jo
 }
 
 // fetchMachine returns details of the machine where the job has been allocated
-func (p *config) fetchMachine(logr logger.Logger, id string) (ip, nodeID string, port int, err error) {
+func (p *config) fetchMachine(logr logger.Logger, id string) (ip, nodeID string, liteEngineHostPort int, ports []int, err error) {
 	// Get the allocation corresponding to this job submission. If this call fails, there is not much we can do in terms
 	// of cleanup - as the job has created a virtual machine but we could not parse the node identifier.
 	l, _, err := p.client.Jobs().Allocations(id, false, nil)
 	if err != nil {
-		return ip, nodeID, port, err
+		return ip, nodeID, liteEngineHostPort, ports, err
 	}
 	if len(l) == 0 {
-		return ip, nodeID, port, errors.New("scheduler: no allocation found for the job")
+		return ip, nodeID, liteEngineHostPort, ports, errors.New("scheduler: no allocation found for the job")
 	}
 
 	nodeID = l[0].NodeID
 	allocID := l[0].ID
 	if nodeID == "" || allocID == "" {
-		return ip, nodeID, port, errors.New("scheduler: could not find an allocation identifier for the job")
+		return ip, nodeID, liteEngineHostPort, ports, errors.New("scheduler: could not find an allocation identifier for the job")
 	}
 
 	alloc, _, err := p.client.Allocations().Info(allocID, &api.QueryOptions{})
 	if err != nil {
-		return ip, nodeID, port, err
+		return ip, nodeID, liteEngineHostPort, ports, err
 	}
 
 	// Not expected - if nomad is unable to find a port, it should not run the job at all.
 	if alloc.Resources.Networks == nil || len(alloc.Resources.Networks) == 0 {
 		err = fmt.Errorf("scheduler: could not allocate network and ports for job")
 		logr.Errorln(err)
-		return ip, nodeID, port, err
+		return ip, nodeID, liteEngineHostPort, ports, err
 	}
 
-	port = alloc.Resources.Networks[0].DynamicPorts[0].Value
+	liteEngineHostPort = alloc.Resources.Networks[0].DynamicPorts[0].Value
+	for i := 1; i < len(alloc.Resources.Networks[0].DynamicPorts); i++ {
+		ports = append(ports, alloc.Resources.Networks[0].DynamicPorts[i].Value)
+	}
 
 	// sanity check
-	if port <= 0 || port > 65535 {
-		err = fmt.Errorf("scheduler: port %d generated is not a valid port", port)
+	if liteEngineHostPort <= 0 || liteEngineHostPort > 65535 {
```

Comment:
Nit: just combine both these into a single loop after iterating over ports and keep the error message
fmt.Errorf("scheduler: port %d generated is not a valid port", port)

Reply:
I didn't get it, how do you combine it into a single loop? 
One option is to iterate over alloc.Resources.Networks[0].DynamicPorts before we assign liteEngineHostPort and the ports slice/array

Reply:
ah ok - this is fine, we can keep as is